src/Platforms/CUDA/CUDAruntime.hpp

```diff
@@ -25,6 +25,17 @@
 #include "Platforms/ROCm/cuda2hip.h"
 #endif
 
+// Work around for clang using gcc/glibc 12 include files in CUDA mode.
```

Comment:
In my understanding, CUDA mode refers compiling CUDA source code by Clang. Here it seems to be a conflict between CUDA host library header files and libstdc++ header files.

Reply:
Yes, the noinline attribute was added to the libstdc++ header files here: https://github.com/gcc-mirror/gcc/commit/dbf8bd3c2f2cd2d27ca4f0fe379bd9490273c6d7#diff-b358f609a31a4af8af72cc3197566abaa157bb7f8681b45580f1e5477540457cR192-R193

Prior to that there were no uses of the noinline attribute in the libstdc++ headers to cause a problem